banderwagon/precomp_multiexp.go

```diff
@@ -1,135 +1,268 @@
 package banderwagon
 
 import (
+	"context"
 	"encoding/binary"
+	"fmt"
 	"io"
 
 	"github.com/crate-crypto/go-ipa/bandersnatch"
 	"github.com/crate-crypto/go-ipa/bandersnatch/fr"
 	"github.com/crate-crypto/go-ipa/common/parallel"
+	"golang.org/x/sync/errgroup"
 )
 
+const (
+	// optimized16BitIdxs is how many elements counting from the first element of the SRS we use a 16-bit table.
+	// For the rest of the (numPoints-optimized16bitIdxs) elements we use a 8-bit table.
+	optimized16BitIdxs = 5
+)
+
+// PrecomputeLagrange contains precomputed tables for a SRS.
 type PrecomputeLagrange struct {
-	inner      []*LagrangeTablePoints
-	num_points int
+	// numPoints is the number of points in the SRS.
+	numPoints int
+	// inner16Bit contains the precomputed tables for the first `optimized16BitIdx` group elements.
+	inner16Bit []*LagrangeTablePoints
+	// inner8Bit contains the precomputed tables for the rest of the group elements.
+	inner8Bit []*LagrangeTablePoints
 }
 
+// Equal returns true if the two PrecomputeLagrange are equal.
 func (pcl PrecomputeLagrange) Equal(other PrecomputeLagrange) bool {
-	if pcl.num_points != other.num_points {
+	if pcl.numPoints != other.numPoints {
```

Comment:
Changed from snake-case to camel case to be coherent with other fields and what's idiomatic in Go. (I think this code might be a port of a Rust version).

Reply:
Nice catch :)